Sources/NIOCore/Codec.swift

```diff
@@ -768,9 +768,11 @@ public final class MessageToByteHandler<Encoder: MessageToByteEncoder>: ChannelO
     private var state: State = .notInChannelYet
     private let encoder: Encoder
     private var buffer: ByteBuffer? = nil
+    private let maxBufferCapacity: Int
```

Comment:
I toyed around the idea of having the ability to switch between different strategies, for example a median of the last N number of write operations and using that as the size to clamp around. I think if there is consistently large requests, this clamping would be more pain than it would be useful. But I don't fully understand the context of the repo so I let it be. LMK your thoughts.

Reply:
I agree that constantly clamping the buffer for requests that are even slightly over the size would have a negative performance impact here. I think we might want to become a bit smarter here to avoid flip flopping the size of the buffer.

Reply:
The easiest strategy is probably to add a power-of-two slack.